plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/service/internal/GroupMetadataServiceImpl.java

```diff
@@ -64,13 +88,60 @@ protected PackageRootIterator doGenerateRegistryRoot(final PackageRequest reques
   protected PackageRoot doGeneratePackageRoot(final PackageRequest request) throws IOException {
     final List<NpmRepository> members = (request.isScoped() &&
         !npmGroupRepository.getId().equals(request.getScope())) ? getScopeMembers(request.getScope()) : getMembers();
-    for (NpmRepository member : members) {
-      final PackageRoot root = member.getMetadataService().generatePackageRoot(request);
-      if (root != null) {
-        return root;
+    if (isMergeMetadata()) {
+      PackageRoot root = null;
+      String latestVersion = null;
+      // apply in reverse order to have "first wins", as package overlay makes overlaid prevail
+      for (NpmRepository member : Lists.reverse(members)) {
+        final PackageRoot memberRoot = member.getMetadataService().generatePackageRoot(request);
+        if (memberRoot != null) {
+          if (root == null) {
+            root = new PackageRoot(npmGroupRepository.getId(), memberRoot.getRaw());
+            latestVersion = getDistTagsLatest(root);
+          }
+          else {
+            String memberLatestVersion = getDistTagsLatest(memberRoot);
+            root.overlayIgnoringOrigin(memberRoot);
+            if (latestVersion != null && memberLatestVersion != null &&
```

Comment:
Can `latestVersion` really be `null` here? it looks like it should have been initialized along with `root`, ie. if `root` is non-null then `latestVersion` should also be non-null?

If on the other hand `latestVersion` can be `null` here then presumably it should be set to `memberLatestVersion` otherwise you could miss a version?